### DISCUSSION TITLE
blog/nextra-3

### DISCUSSION THREAD

**Discussion by giscus** [the-guild-org/the-guild-website#1488] [Docs Discussions]
# blog/nextra-3

MDX 3, new i18n, new _meta files with JSX support, more powerful TOC, remote MDX, better bundle size, MathJax, new code block styles, shikiji, ESM-only and more

https://the-guild.dev/blog/nextra-3

**Comment by abenhamdine** [comment]
Great news !
Congrats for all this stuff.

**Comment by SaadBazaz** [comment]
Been waiting for this for too long <3
However, would really really need appDir support. Bumping up to >= Next 13 does not make sense if we can't use the new router; we're pretty much stuck at Next 13 because 14 doesn't even have the `pages` dir. 

Need this in appDir because I want to build a hybrid site which can have both docs and SaaS.

**Reply by dimaMachina** [reply]
appDir support will be the main feature for v4

> Bumping up to >= Next 13 does not make sense if we can't use the new router; we're pretty much stuck at Next 13 because 14 doesn't even have the pages dir

this is wrong, you can use both routers in Next.js, `pages` router is not deprecated and not removed from core Next.js

**Reply by raulmarindev** [reply]
You can even combine them and have some pages/api routes inside the page folder and others in the app folder.

**Comment by Hidde-Heijnen** [comment]
Love this Dima! Glad there is a future for this amazing library and it is in good hands with you working on it!

**Comment by wnqueiroz** [comment]
Excited for the launch! Congratulations on the work Dimitri 🚀.
Will v3 resolve the issue in indexing/searching remote MDX content (`<RemoteContent />`)? 👀

**Reply by dimaMachina** [reply]
V4 will indexes remote content with migration from flexsearch to pagefind

**Comment by whitedark9** [comment]
Thanks for the new version. I am not a professional programmer. Has anyone tried to use the alpha version? I can't click on the headings of TOC. Is it just me?

**Comment by BalintBanyasz** [comment]
Thanks for the new version. It looks like `pageOpts.route` has been removed as well. I currently use this property in my `theme.config.tsx` file to add a custom footer with social media share buttons. Is there a way to access the route via `useBlogContext()` in the new version? `pageMap` returns the route for all pages, but I would need to access the route of the current page.

**Reply by 87xie** [reply]
Not sure if this helps you.

```ts
import { useRouter } from 'nextra/hooks'

const { route } = useRouter()
```

**Reply by BalintBanyasz** [reply]
> Not sure if this helps you.
> 
> ```ts
> import { useRouter } from 'nextra/hooks'
> 
> const { route } = useRouter()
> ```

Thanks for the tip, it pointed me in the right direction. The documentation has a detailed [example](https://nextra.site/docs/docs-theme/theme-configuration#dynamic-tags-based-on-page) of using dynamics tags, which worked well for me.

**Comment by mackenly** [comment]
Replacing the old `useNextSeoProps` and running into some confusion. The [example](https://nextra.site/docs/docs-theme/theme-configuration#dynamic-tags-based-on-page) in the docs show what is called a *functional component* which seems straight forward enough, but how is this used within the `theme.config.jsx`? Thanks!

Edit:
You just add it straight into the config. Place any imports at the top of the file and then inline the function like this:
```javascript
import { useConfig } from 'nextra-theme-docs';

export default {
  // config things here...
  head() {
    const { frontMatter } = useConfig();

    return (
      <>
        <title>{frontMatter.title || 'My Project' + ' - Docs'}</title>
        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <meta name="description" content="Desc goes here" />
        <meta property="og:title" content={frontMatter.title || 'Docs' + ' - My Project Docs'} />
        <meta property="og:description" content="Desc goes here" />
        <meta name="msapplication-TileColor" content="#000000" />
        <meta name="theme-color" content="#000000" />
      </>
    );
  },
  // ... other things here
}
```

**Comment by raulmarindev** [comment]
Amazing work! Thanks 💗

**Comment by varlord20** [comment]
Nice work!
